pkg/types/builder.go

```diff
@@ -96,27 +99,58 @@ func (g *Builder) buildResource(res *schema.Resource, refs config.References, na
 		if comment.TerrajetOptions.FieldJSONTag != nil {
 			jsonTag = *comment.TerrajetOptions.FieldJSONTag
 		}
-		fieldType, err := g.buildSchema(sch, refs, append(names, fieldName.Camel))
+
```

Comment:
During the initial implementation, I was thinking to have a generic interface like `FieldProcessor` that we can call in a for loop here to process each field in the schema (like `Reference`, `Sensitive`, `Generic` etc) so that each behavior logic would be contained in its own field processor but didn't pursue it. It may not even be possible since there are a few independent things each processor would do and it could encompass more than one field.

Not related to this PR at all, but I'm curious about your thoughts about whether it'd be possible to do it at some point given that you have more context about the nature of the flow here.

Reply:
Yeah, I think that's would improve the readability and maintenance of this path of the codebase. I was also not feeling good about adding new lines to a single flow. Not sure how much we would need to add though.
This could be tried out while writing unit tests here may be.

Reply:
Could you leave a comment on that issue or open another issue so that we can track it?

Reply:
https://github.com/crossplane-contrib/terrajet/issues/17